### PR TITLE
Implement use of step.id

### PR DESCRIPTION
**Pull request by autt**
Resolves #31 

This PR is dependent on [this ISAR PR](https://github.com/equinor/isar/pull/114)

### CODE REVIEW

File: src/isar_turtlebot/robotinterface.py [hunk 1]
```diff
@@ -49,15 +50,17 @@ def __init__(self):
...
+    def schedule_task(self, task: Task) -> Tuple[bool, Optional[Joints]]:
+        self.id = self._get_turtlebot_id()
+        self._publish_task(task=task)
+        return True, None
 
     def mission_scheduled(self) -> bool:
         return False
 
-    def mission_status(self, mission_id: Any) -> MissionStatus:
+    def mission_status(self, mission_id: UUID) -> MissionStatus:
```
Comment (knord94) [inline src/isar_turtlebot/robotinterface.py:63]:
```suggestion
    def mission_status(self) -> MissionStatus:
```

Implement mission_id as optional input in ISAR if it is necessary for some robots?

File: src/isar_turtlebot/robotinterface.py [hunk 2]
```diff
@@ -49,15 +50,17 @@ def __init__(self):
         )
         self.current_task: Optional[str] = None
         self.inspection_status: Optional[TurtlebotStatus] = None
+        self.id: str = ""
 
-    def schedule_task(self, task: Task) -> Tuple[bool, Optional[Any], Optional[Joints]]:
-        run_id: str = self._publish_task(task=task)
-        return True, run_id, None
+    def schedule_task(self, task: Task) -> Tuple[bool, Optional[Joints]]:
```
[the remark this answers is not in the document]
Comment (knord94) [reply src/isar_turtlebot/robotinterface.py:55]: Resolved in [#28](https://github.com/equinor/isar/issues/28)

File: src/isar_turtlebot/robotinterface.py (hunk 1 above)
Comment (autt) [reply src/isar_turtlebot/robotinterface.py]: Good idea :+1:

File: src/isar_turtlebot/robotinterface.py [hunk 3]
```diff
@@ -262,27 +248,21 @@ def _publish_navigation_task(self, pose: Pose) -> None:
...
-                )
-            current_run_id = self._get_run_id()
 
-        return current_run_id
+            if execution_time > self.inspection_task_timeout:
+                self.inspection_status = TurtlebotStatus.Failure
+                raise TimeoutError(f"Storing image for TurtleBot3 timed out.")
+
+    def _get_turtlebot_id(self) -> Optional[str]:
+        status_msg: dict = self.bridge.mission_status.get_value()
```
Comment (knord94) [inline src/isar_turtlebot/robotinterface.py:262]:
```suggestion
        status_msg: dict = self.bridge.task_status.get_value()
```

**Review by knord94 — CHANGES_REQUESTED** [review]

File: src/isar_turtlebot/robotinterface.py [hunk 4]
```diff
@@ -49,67 +50,64 @@ def __init__(self):
         )
         self.current_task: Optional[str] = None
         self.inspection_status: Optional[TurtlebotStatus] = None
+        self.id: str = ""
```
Comment (vetlek) [inline src/isar_turtlebot/robotinterface.py:53]: It is not obviously clear what the id attribute is referring to. Based on the code lower down it is the id of the current active task, but that is not clear based on the name id.

File: src/isar_turtlebot/robotinterface.py (hunk 3 above)
Comment (knord94) [reply src/isar_turtlebot/robotinterface.py:262]: The same applies for `navigation_status()`, line 173

File: src/isar_turtlebot/robotinterface.py [hunk 5]
```diff
@@ -262,27 +248,21 @@ def _publish_navigation_task(self, pose: Pose) -> None:
...
-                    f"Scheduling of task for TurtleBot3 timed out. Run ID: {current_run_id}"
-                )
-            current_run_id = self._get_run_id()
 
-        return current_run_id
+            if execution_time > self.inspection_task_timeout:
+                self.inspection_status = TurtlebotStatus.Failure
+                raise TimeoutError(f"Storing image for TurtleBot3 timed out.")
+
+    def _get_turtlebot_id(self) -> Optional[str]:
```
Comment (vetlek) [inline src/isar_turtlebot/robotinterface.py:261]: Change to `_get_turtlebot_task_id(self)` to make it  clear that we are getting the id of the navigation task on the robot?

File: src/isar_turtlebot/robotinterface.py [hunk 6]
```diff
@@ -49,67 +50,64 @@ def __init__(self):
...
         self, inspection: Inspection
     ) -> Optional[InspectionResult]:
         if isinstance(inspection, ImageReference):
             try:
                 image_data = self.bridge.visual_inspection.read_image(
-                    run_id=inspection.id
+                    inspection_id=inspection.id
                 )
 
                 inspection_result = Image(
```
Comment (knord94) [inline src/isar_turtlebot/robotinterface.py:113]: "self.logger.error" on line 116 and line 136

File: src/isar_turtlebot/robotinterface.py (hunk 5 above)
Comment (autt) [reply src/isar_turtlebot/robotinterface.py:261]: This is a good catch, agree it is not clear which id this is. I renamed it to `self.goal_id`, as it is not the same as `task.id`. Other suggestions are welcome.

**Review by vetlek — APPROVED** [review]
LGTM

**Review by knord94 — APPROVED** [review]

**Pull request merged**